internal/preflight/preflight.go

```diff
@@ -17,33 +20,37 @@ const (
 	VerificationStatusReasonBuildConfigPresent = "Verification successful, all driver-containers have paired BuildConfigs in the recipe"
 	VerificationStatusReasonNoDaemonSet        = "Verification successful, no driver-container present in the recipe"
 	VerificationStatusReasonUnknown            = "Verification has not started yet"
-	VerificationStatusReasonVerified           = "Verification successful, this Module would be verified again in this Preflight CR"
+	VerificationStatusReasonVerified           = "Verification successful (%s), this Module would not be verified again in this Preflight CR"
 )
 
-//go:generate mockgen -source=preflight.go -package=preflight -destination=mock_preflight_api.go
+//go:generate mockgen -source=preflight.go -package=preflight -destination=mock_preflight_api.go PreflightAPI, preflightHelperAPI
 
 type PreflightAPI interface {
-	PreflightUpgradeCheck(ctx context.Context, mod *kmmv1beta1.Module, kernelVersion string) (bool, string)
+	PreflightUpgradeCheck(ctx context.Context, pv *kmmv1beta1.PreflightValidation, mod *kmmv1beta1.Module, kernelVersion string) (bool, string)
 }
 
 func NewPreflightAPI(
 	client client.Client,
+	buildAPI build.Manager,
 	registryAPI registry.Registry,
+	statusUpdater statusupdater.PreflightStatusUpdater,
 	kernelAPI module.KernelMapper) PreflightAPI {
+	helper := newPreflightHelper(client, buildAPI, registryAPI)
```

Comment:
i disagree. Originally those functions were private functions of the preflight, i moved them into private interface just for unit-tests. Registry should deal only with accessing the registry and the image contents, the logic of how to check and what to check should be in preflight. The same goes for verifyBuild, and in later PRs it will be epanded more with Push option and also sign verification

Reply:
> Registry should deal only with accessing the registry and the image contents, the logic of how to check and what to check should be in preflight.

I agree with it. In that case, how about a registry API function that gets a predicate function dictated by the preflight code? That way, you keep the registry logic in the registry package, and you extract the preflight part to be injected as an argument to the registry package.

Regarding `verifyBuild`, it only have 1 API call and it's just checking the return values, meaning that having it as a private method will look exactly the same regarding mocking (instead of mocking `verifyBuild` we will mock `Sync`.

Reply:
1) i don't like the idea of predicate function passed as argument. It makes our code much less readable. And it does not save us anything, it is still needs to be implemented in a calling package. 
2) personally i would prefer not to use private interfaces at all and just write more elaborate unit-test for private functions, but i think we agreed that this discussion will wait until we reach a stage where we try to refactor the module reconciler. As for verifyBuiild - this function will probably be expended in later PRs with push configuration and sign verification, in which case it will have more then 1 mock call. And since we already have private interface that includes verifyImage, adding verifyBuild just make code structure more logical

Reply:
> And it does not save us anything

It is maintaining a better separation between the logical packages IMO.

Ack on `verifyBuild`. I would still move the registry functionality to the `registry` package with some predicate (that contains the logic which isn't related to the registry package, like we do when we pass the authenticator) but seems like we are not going to agree on it so let's merge :)